From e14f3dc392daed10ab3186957ffe000a4463167d Mon Sep 17 00:00:00 2001
From: meta-calculinux <meta-calculinux@example.com>
Date: Sat, 21 Feb 2026 00:00:00 +0000
Subject: [PATCH] fix: modernize autoconf/automake macros and build system

Replace deprecated autoconf/automake macros that fail with modern toolchains:
- AM_CONFIG_HEADER -> AC_CONFIG_HEADERS
- AM_INIT_AUTOMAKE with positional args -> AM_INIT_AUTOMAKE([foreign])
- Remove AM_MAINTAINER_MODE (breaks autoreconf)
- Remove AC_C_CONST and AC_TYPE_SIGNAL (removed in autoconf 2.70+)

Fix Makefile.am issues for out-of-tree builds:
- dist_libgraph_DATA and dist_libgraphfont_DATA used globs (doc/*.html,
  Font/*) which automake does not expand; replace with explicit lists using
  $(srcdir)/... so out-of-tree builds can locate the source files.
- pkgconfig install used PROGRAMS primary for a shell-script-generated .pc
  file; use DATA primary and an explicit rule with 'sh' instead.
---
 Makefile.am  | 26 ++++++++++++++++----------
 configure.in | 11 ++++-------
 2 files changed, 20 insertions(+), 17 deletions(-)

--- a/configure.in	2005-12-12 04:32:57.000000000 -0500
+++ b/configure.in	2026-02-21 21:59:45.667616093 -0500
@@ -4,9 +4,8 @@
 dnl Check system type
 AC_CANONICAL_TARGET
 
-AM_CONFIG_HEADER(config.h)
-AM_INIT_AUTOMAKE("libgraph", "1.0.2")
-AM_MAINTAINER_MODE
+AC_CONFIG_HEADERS([config.h])
+AM_INIT_AUTOMAKE([foreign])
 
 # Checks for programs.
 AC_PROG_CC
@@ -88,13 +87,11 @@
 AC_CHECK_HEADERS([fcntl.h stdlib.h string.h termios.h unistd.h])
 
 # Checks for typedefs, structures, and compiler characteristics.
-AC_C_CONST
 AC_TYPE_PID_T
 
 # Checks for library functions.
 AC_FUNC_FORK
 AC_FUNC_MALLOC
-AC_TYPE_SIGNAL
 AC_FUNC_VPRINTF
 AC_CHECK_FUNCS([atexit putenv rint strdup])
 
--- a/Makefile.am	2005-12-12 04:43:39.000000000 -0500
+++ b/Makefile.am	2026-02-21 21:59:45.667560829 -0500
@@ -22,8 +22,32 @@
 
 libgraphdir = $(pkgdatadir)
 libgraphfontdir = $(libgraphdir)/Font
-dist_libgraph_DATA = doc/*.html
-dist_libgraphfont_DATA = Font/*
+dist_libgraph_DATA = \
+	$(srcdir)/doc/index.html \
+	$(srcdir)/doc/Introduction.html \
+	$(srcdir)/doc/Function_List.html \
+	$(srcdir)/doc/Developer.html \
+	$(srcdir)/doc/Credits.html \
+	$(srcdir)/doc/Bugs.html \
+	$(srcdir)/doc/COPYING.html
+dist_libgraphfont_DATA = \
+	$(srcdir)/Font/black.png \
+	$(srcdir)/Font/blue.png \
+	$(srcdir)/Font/brown.png \
+	$(srcdir)/Font/cyan.png \
+	$(srcdir)/Font/darkgray.png \
+	$(srcdir)/Font/green.png \
+	$(srcdir)/Font/icon.bmp \
+	$(srcdir)/Font/lightblue.png \
+	$(srcdir)/Font/lightgray.png \
+	$(srcdir)/Font/lightgreen.png \
+	$(srcdir)/Font/lightmagenta.png \
+	$(srcdir)/Font/lightred.png \
+	$(srcdir)/Font/magenta.png \
+	$(srcdir)/Font/red.png \
+	$(srcdir)/Font/smstone16.png \
+	$(srcdir)/Font/white.png \
+	$(srcdir)/Font/yellow.png
 
 
 
@@ -56,11 +80,8 @@
 ## pkgconfig .pc file
 
 libgraphpkgdir = $(libdir)/pkgconfig
-libgraphpkg_PROGRAMS = libgraph.pc
-libgraph_pc_SOURCES = pcgen.sh
-
-$(libgraphpkg_PROGRAMS): $(libgraph_pc_SOURCES)
-	./$< $(prefix) $(libdir) $(bindir) $(includedir) $(PACKAGE) $(VERSION) > $@; 
-
-CLEANFILES = *.pc
-endif 
+libgraphpkg_DATA = libgraph.pc
+libgraph.pc: pcgen.sh
+	sh $< $(prefix) $(libdir) $(bindir) $(includedir) $(PACKAGE) $(VERSION) > $@
+CLEANFILES = libgraph.pc
+endif
